lib/engine/share_price.rb

```diff
@@ -2,7 +2,8 @@
 
 module Engine
   class SharePrice
-    attr_reader :coordinates, :price, :corporations, :can_par, :type, :types
+    attr_reader :coordinates, :price, :can_par, :type, :types
```

Comment:
i don't this change is necessary, you can just mutate corporations

corporations.clear
corporations.concat(reordered)

Reply:
good point